Localtuya 4.0 (#829)
rospogrigio/localtuya: info.md

![logo](https://github.com/rospogrigio/localtuya-homeassistant/blob/master/img/logo-small.png)

A Home Assistant custom Integration for local handling of Tuya-based devices.

> This custom integration updates device status via push updates instead of polling, so status updates are fast (even when manually operated).

Reviewer: ```suggestion
This custom integration updates device status via pushing updates instead of polling, so status updates are fast (even when manually operated).
```

# Usage:

**NOTE: You must have your Tuya device's Key and ID in order to use LocalTuya. The easiest way is to configure the Cloud API account in the integration. If you choose not to do it, there are several ways to obtain the local_keys depending on your environment and the devices you own. A good place to start getting info is https://github.com/codetheweb/tuyapi/blob/master/docs/SETUP.md .**

**NOTE 2: If you plan to integrate these devices on a network that has internet and blocking their internet access, you must also block DNS requests (to the local DNS server, e.g. 192.168.1.1). If you only block outbound internet, then the device will sit in a zombie state; it will refuse / not respond to any connections with the localkey. Therefore, you must first connect the devices with an active internet connection, grab each device localkey, and implement the block.**

# Adding the Integration

> **NOTE: starting from v.4.0.0, configuration using YAML files is no longer supported. The integration can be configured only using the config flow.**

Reviewer: ```suggestion
**NOTE: starting from v4.0.0, configuration using YAML files is unsupported. The integration can only be configured using the config flow.**
```

If you had configured LocalTuya using YAML files, you can delete all its references from within the YAML files because they will no longer be considered so they might bring confusion (only the logger configuration part needs to be kept, of course, see [Debugging](https://github.com/rospogrigio/localtuya#debugging) ).

# Energy monitoring values

You can obtain Energy monitoring (voltage, current) in two different ways:

1) Creating individual sensors, each one with the desired name.
  Note: Voltage and Consumption usually include the first decimal. You will need to scale the parament by 0.1 to get the correct values.

> 1) Access the voltage/current/current_consumption attributes of a switch, and define template sensors

Reviewer: ```suggestion
2) Access the voltage/current/current_consumption attributes of a switch, and define template sensors
```

Unsure if this is a markdown rendering behaviour, or this should be 1); 2); 3)
